debezium-server-pulsar/src/main/java/io/debezium/server/pulsar/PulsarChangeConsumer.java

```diff
@@ -152,4 +175,52 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
         }
         committer.markBatchFinished();
     }
+
+    @SuppressWarnings("unchecked")
+    private void handleBatchAsync(List<ChangeEvent<Object, Object>> records, RecordCommitter<ChangeEvent<Object, Object>> committer) throws InterruptedException {
+        List<CompletableFuture<MessageId>> futures = new ArrayList<>();
+
+        for (ChangeEvent<Object, Object> record : records) {
+            final TypedMessageBuilder message = buildMessage(record);
+
+            final ChangeEvent<Object, Object> currentRecord = record;
+
+            CompletableFuture<MessageId> future = message
+                    .sendAsync()
+                    .thenAccept(messageId -> {
+                        LOGGER.trace("Sent message with id: {}", messageId);
+                        try {
+                            committer.markProcessed(currentRecord);
+                        }
+                        catch (InterruptedException e) {
+                            throw new DebeziumException(e);
+                        }
+                    })
+                    .exceptionally(ex -> {
+                        Throwable exception = (Throwable) ex;
+                        LOGGER.error("Failed to send record to {}:", record.destination(), exception);
+                        throw new DebeziumException(exception);
+                    });
+
+            futures.add(future);
+        }
+
+        try {
+            // Wait for all CompletableFutures to complete
+            CompletableFuture
+                    .allOf(futures.toArray(new CompletableFuture[0]))
+                    .join();
```

Comment:
I'd recommend to provide a time-bound solution too - see https://issues.redhat.com/browse/DBZ-5175?focusedId=20451392&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20451392 for an issue we have in Google PubSub sink so better to be ready for that too. The timeout would be configurable with default 0 meaning unlimited.

Reply:
Done. I have created a new configuration property for it, called `timeout`. Is it okay this way?